internal/components/processors_test.go

```diff
@@ -199,6 +200,33 @@ func verifyProcessorLifecycle(t *testing.T, factory processor.Factory, getConfig
 	}
 }
 
+// verifyProcessorShutdown is used to test if a processor type can be shutdown without being started first.
+// We disregard errors being returned by shutdown, we're just making sure the processors don't panic.
+func verifyProcessorShutdown(tb testing.TB, factory processor.Factory, getConfigFn getProcessorConfigFn) {
+	ctx := context.Background()
+	processorCreationSet := processortest.NewNopCreateSettings()
+
+	if getConfigFn == nil {
+		getConfigFn = factory.CreateDefaultConfig
+	}
+
+	createFns := []createProcessorFn{
+		wrapCreateLogsProc(factory),
+		wrapCreateTracesProc(factory),
+		wrapCreateMetricsProc(factory),
+	}
+
+	for _, createFn := range createFns {
+		p, err := createFn(ctx, processorCreationSet, getConfigFn())
+		if errors.Is(err, component.ErrDataTypeIsNotSupported) {
+			continue
+		}
+		assert.NotPanics(tb, func() {
+			_ = p.Shutdown(ctx)
```

Comment:
Any reason not to check the error here?

Reply:
The memory limiter processor, which is in core, returns an error. We'd need to change it first before asserting here.